src/components/map/MapService.js

```diff
@@ -240,6 +244,10 @@ goog.require('ga_urlutils_service');
           updateDelay: {
             writable: true,
             value: null
+          },
+          userVisible: {
+            writable: true,
+            value: olLayer.getVisible()
```

Comment:
Not only KML I just saw that concerns also with External WMS.  That concerns all layers which set initial visibility with the layer's constructor and not with the overrided setVisible method.